probe-rs/src/debug/debug_info.rs

```diff
@@ -1120,6 +1121,35 @@ impl DebugInfo {
     }
 }
 
+/// This is a fuzzy 'equality' check for path names.
+/// If both paths are absolute, then it will use the standard `Path::eq` method.
+/// If the secondary path is relative, then it will succeeed if the primary ends with the compared path,
+/// e.g. `/home/user/project/src/main.rs` ends with `src/main.rs` and will return true.
```

Comment:
What's the reasoning behind this? I'm not a big fan of trying to match files somehow, and would prefer if we keep the current logic.

Reply:
@Tiwalun, the reason is that VSCode passes in an absolute path to test against, and the C makefiles often uses relative files. Without this, the success rate of resolving C source locations between what DAP request has versus what debug_info has, was literally ZERO in the test case I had. That is why I coded it to short_circuit to existing behaviour `primary_path.eq(secondary_path)`. I can undo it, but then the test case from the original poster no longer works at all, and the PR becomes obsolete.

Reply:
See new implementation please.